io/src/main/scala/sbt/io/IO.scala

```diff
@@ -1177,7 +1177,7 @@ object IO {
       else
         new URI(str + "/")
 
-    dirURI.normalize
+    new File(dirURI).toPath.normalize.toUri
```

Comment:
The creation of the first URI object seems wasteful here. See also https://github.com/sbt/io/pull/132, which states that URI object can lead to hundreds of MB in heap.

Reply:
Shall investigate this once I regain access to a Windows Machine.

Indeed there is something fishy there. If `dirURI` follows u2 notation, then calling `dirURI.normalize` should be correct. While I think the other changes in the PR makes sense, this particular change is just a bandaid that somehow fixes the sbt crash without fixing the underlying issue.

I now actually suspect the hidden issue might be with `IO.toURL`, which is called by `ProjectRef.apply` on sbt side. Shall verify the hypothesis once I get the windows machine.

Reply:
Btw going slightly off topic but your [blogpost](https://eed3si9n.com/encoding-file-path-as-URI-reference/) about file URI schema is so helpful!